### ISSUE TITLE
Incorrect total available bandwidth from astats

### ISSUE THREAD

**Issue by jhg03a** [apache/trafficcontrol#4023]
## I'm submitting a ...

- [ ] bug report
- [ ] new feature / enhancement request
- [x] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [x] ATS astats plugin
- [ ] unknown

## Current behavior:

Currently the value astats returns for the network link speed of the nic on normal nics is correct.  However when using a bonded interface, this number always reflects the total aggregate speed without regard to link status.  This can lead to situations where a bond can be partially failed, but Traffic Monitor is unaware that the available bandwidth capacity has changed and therefore keep a cache as healthy when no bandwidth is legitimately available.

## Expected / new behavior:

I expect the link speed from astats on a bonded interface to reflect the sum of healthy links rather than the simple aggregate.

## Minimal reproduction of the problem with instructions:

1. Install ATS with the astats plugin on a host that uses two links in a bond
2. Observe the bandwidth capacity being reported is the sum of the two links
3. Down one of the links in the bond
4. Observe the bandwidth capacity being reported doesn't change

**Comment by ezelkow1** [comment]
There currently exists the system_stats.so plugin.  This will inject the stats values for every available interface on a box, as well as listing the slaves for any bonds on bonded interfaces. Via this plugin it should be possible to calculate everything you want

**Comment by zrhoffman** [comment]
@jhg03a, does the `system_stats.so` plugin solution work for you?

**Comment by jhg03a** [comment]
No. The system stats plugin also doesn't take this into account I believe.

**Comment by ezelkow1** [comment]
> No. The system stats plugin also doesn't take this into account I believe.

But if you have all the information about the various links and link states then you should be able to figure it out outside of the plugin right? Thats what I was getting at by just having all the info you can calculate what you want

**Comment by jhg03a** [comment]
There's nowhere to do that because TM just expects a single piece of data to evaluate via the rascal profile
